cmake/kokkos_arch.cmake

```diff
@@ -229,9 +229,14 @@ IF(KOKKOS_ARCH_NATIVE)
     MESSAGE(FATAL_ERROR "MSVC doesn't support ARCH_NATIVE!")
   ENDIF()
 
+  IF(CMAKE_SYSTEM_PROCESSOR MATCHES "(x86)|(X86)|(amd64)|(AMD64)")
```

Comment:
Consider making the string all uppercase before to avoid vexing `(Amd64)` not matching